vue-components/src/components/Lookup.vue

```diff
@@ -86,6 +92,41 @@ export default Vue.extend( {
 			},
 			default: 'medium',
 		},
+		/**
+		 * Sets the value of the Lookup component's inner `<input>` element. This prop can be used with the `.sync`
+		 * modifier. When bound to a field in the consuming component's data object, it can be used within a watcher or
+		 * computed property to dynamically update the Lookup's `menuItems` prop.
+		 */
+		searchInput: {
+			type: String,
+			default: '',
+		},
+	},
+
+	methods: {
+		onInput( value: string ): void {
+			this.hasItemSelected = false;
+
+			// the following comment generates the event's description for the docs tab in storybook
+			/**
+			 * Enables the `searchInput` prop to be used with the `.sync` modifier. It's used to transport the value of
+			 * the Lookup component's inner `<input>` element to the parent component.
+			 */
+			this.$emit( 'update:searchInput', value );
+		},
+
+		onSelect( menuItem: MenuItem ): void {
+			this.hasItemSelected = true;
+
+			// the following comment generates the event's description for the docs tab in storybook
+			/**
+			 * This even is emitted whenever an item is selected on the Lookup. The event payload contains the whole
+			 * MenuItem object.
+			 */
+			this.$emit( 'input', menuItem );
```

Comment:
Fantastic choice! :sunglasses: we shouldn't care whether the implementer of this components has a `value` key or not....